app/tests/archives_tests/test_serializers.py

```diff
@@ -26,13 +27,16 @@
                     "algorithms",
                     "logo",
                     "description",
+                    "hanging_protocol",
                 ),
             },
         )
     ),
 )
 class TestSerializers:
     def test_serializer_valid(self, serializer_data, rf):
+        hp = HangingProtocolFactory(json=([{"viewport_name": "main"}]))
+        serializer_data["factory"] = ArchiveFactory(hanging_protocol=hp)
```

Comment:
To make this work correctly I think you need remove these lines and use a subclassed version of ArchiveFactory on line 19 where the hanging protocol field is defined.